index.html

```diff
@@ -1302,6 +1302,15 @@ <h2>Public Keys</h2>
 the <a>DID document</a>; see Section <a href="#authorization-and-delegation"></a>.
       </p>
 
+      <p>
+The value of the <code>id</code> property MAY be structured as a <a href="https://en.wikipedia.org/wiki/Compound_key">compound key</a>. 
+This is especially useful for integrating with existing key management systems and key formats such as <a href="https://tools.ietf.org/html/rfc7517">JWK</a>.
+It is RECOMMENDED that <a href="https://tools.ietf.org/html/rfc7517">JWK</a> <code>kid</code> values are set to the <a href="https://tools.ietf.org/html/rfc7638">public key fingerprint</a>. 
+It is RECOMMENDED that verification methods that use JWKs to represent their 
```

Comment:
This second recommendation surprises me. I like it, but it surprises me. The reason is that most examples I've seen use `id` instead -- and indeed, that seems to jive with what the relative URL language assumes.

Everything else in a DID document will use `id` as the fragment identifier. The `id` field is intentionally NOT pointing to immutable content; that is, by design, we want did:example:abc#key1 to be able to point to various, different values over its lifespan. I detest this design choice, considered it a mistake, and argued against it, but was overruled.

Recommending that we make the `kid` property the fragment identifier for a JWK would be great, except that it's so at odds with everything else we're doing in the doc.

Reply:
If we keep this, can we *also* recommend that we NOT use an `id` property in the same JWK? It seems very messy to me to have a JWK have both, and yet to have its fragment identifier derive from `kid`. A recipe for errors.

Reply:
The `id` attribute is for the verification method, the JWK is the value of `publicKeyJwk`. IMO, it's best that we keep consistency with `id` for verification methods. There's a desire here for some linkage, if possible, between `kid` and `id`, but it may not always be possible.

Reply:
indeed, per the RFC linked... `kid` can be whatever people want... just like `id`.... https://tools.ietf.org/html/rfc7638#section-3.1

`fingerprint` or `kid` are used in the existing systems... `verificationMethod.id` is used with resolvers... we need these to be different or we are going to lots of problems... here is a more complete example of publicKeyJwk integration with JWKS...

https://github.com/transmute-industries/lds-jws2020/blob/master/docs/example/didDocJwks.json

https://github.com/transmute-industries/lds-jws2020/blob/master/docs/example/didDoc.json

You can imagine that you might use the same JWKS with multiple DIDs... if you chose to implement a custom `kid`... now not only can you not use the caching properties associated with it... but you can't actually support 2 did documents with the same JWKS....

Thats why we are trying to recommend that they follow the convention of the following format (regardless of key representation)... 

`did:example:123#fingerprint`... in the case of did:key with publicKeyBase58, the fingerprint is calculated from multibase / multcodec... in the case of publicKeyJwk, its per the RFC...

@dlongley maybe it would help to add the recommendation regarding  publicKeyBase58?